content/en/blog/_posts/2024-01-26-webassembly-on-kubernetes-the-practice-guide/index.md

```diff
@@ -0,0 +1,845 @@
+
+```bash
+curl http://localhost:8080
+```
+
+The output is:
+
+```bash
+Hello, World!
+```
+
+To delete the container, you can execute the following command:
+
+```bash
+crun kill wasm-demo-app SIGKILL
+```
+
+### Run Wasm modules via high-level container runtimes
+
+The _container shim_ serves as a bridge between high-level and low-level container runtimes. Its main purpose is to abstract low-level runtime details, enabling uniform management of various low-level runtimes in high-level runtime. In this section, I will use _[containerd](https://github.com/containerd/containerd)_ as an example. Containerd is an industry-standard container runtime with an emphasis on simplicity, robustness, and portability. 
+
+Containerd can manage Wasm modules in two ways:
+1. Manages Wasm modules through container runtimes like crun and youki that support building with the Wasm runtime library. These two runtimes can also run regular Linux containers. Containerd uses `containerd-shim-runc-v2` to interface with low-level container runtimes.
+1. Manages Wasm modules directly through Wasm runtimes, such as Slight, Spin, WasmEdge, and Wasmtime. Containerd uses `containerd-wasm-shim(runwasi)` to interface with Wasm runtimes.
+
+#### Containerd + Crun
+
+In this section, I will demonstrate how to configure crun as runtime in containerd, enabling support for running Wasm modules.
+
+{{< note >}}
+Ensure that crun binary with Wasm support has been installed as per the instructions in the section: [Run Wasm modules via low-level container runtimes](#run-wasm-modules-via-low-level-container-runtimes).
+{{< /note >}}
+
+Run the following commands to install containerd:
+
+```shell
+export VERSION="1.7.3"
+sudo apt install -y libseccomp2
+sudo apt install -y wget
+
+wget https://github.com/containerd/containerd/releases/download/v${VERSION}/cri-containerd-cni-${VERSION}-linux-amd64.tar.gz
+wget https://github.com/containerd/containerd/releases/download/v${VERSION}/cri-containerd-cni-${VERSION}-linux-amd64.tar.gz.sha256sum
+# expected checksum: ea70faeb6c5d656fa0787dfc7d88a48daf961482c46bb22953cb5396289fd5b8
+sha256sum --check cri-containerd-cni-${VERSION}-linux-amd64.tar.gz.sha256sum
+
+sudo tar --no-overwrite-dir -C / -xzf cri-containerd-cni-${VERSION}-linux-amd64.tar.gz
+sudo systemctl daemon-reload
+sudo systemctl enable containerd --now
+```
+
+You can run Wasm modules through containerd:
+- `--runc-binary`: Specifies to use `crun` as low-level runtime to start the container.
+- `--runtime`: Specifies the version and name of the shim, which are converted by containerd into the binary name of the shim. For example: `io.containerd.runc.v2` → `containerd-shim-runc-v2`. Containerd starts the shim by running the `containerd-shim-runc-v2` binary file, which subsequently invokes `crun` to launch the container.
+- `--label`: Adds `"module.wasm.image/variant": "compat"` to indicate that this is a Wasm application without a guest OS.
```

Comment:
I'm worried about plagiarism. This seems similar to https://bitone.dev/practical-guide-to-webAssembly-in-cloud-native/

Can you explain the similarity?